### PR TITLE
Set up analytics tracking with ahoy

### LINKED ISSUE #720: Install and configure ahoy for metrics collection
### proposal
[Ahoy](https://github.com/ankane/ahoy) provides a simple first-party analytics solution that gives us more control than google analytics.

- The implementation of metrics collection for PURLs using ahoy has two pieces: this is the server-side piece.
- The work to actually collect metrics will happen on the client side in sul-embed, using [ahoy.js](https://github.com/ankane/ahoy).
- Ahoy sets up a new API in PURL to which embedded apps can report analytics.
- Once it is installed, events (in the form of `Event` objects) are saved to a database (note that Purl doesn't currently have a db).

view the [requirements analysis of ahoy](https://docs.google.com/document/d/1W5rmHTMRGso5uxqIKqptvUYVYIn9cUeo6Nm9rx96XTU/edit#heading=h.noqmyhw0ebos) in the requirements doc.

### steps

- [x] install the ahoy gem
- [x] kick the tires on the API, maybe using `curl`, to see if events can be created and tied to a `PurlResource`
- [x] make sure ahoy does the right thing in local dev vs production-like environments

### PR DESCRIPTION
**Pull request by thatbudakguy**
Closes #720 

- Set up Ahoy for analytics tracking
- Set up a SQLite database for tracking analytics
- Add view and download tracking accessors to PurlResource

See the changes to the README for an explanation of how to fake analytics events in local development.

### CODE REVIEW

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -40,6 +40,27 @@
   # instead of true.
   config.use_transactional_fixtures = false
```
Comment (jcoyne) [inline spec/rails_helper.rb:41]: Can't you just set this to `true` and then you don't need `DatabaseCleaner`?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: config/database.yml [hunk 2]
```diff
@@ -1,22 +1,25 @@
...
 test:
-  adapter: sqlite3
+  <<: *default
   database: db/test.sqlite3
-  pool: 5
-  timeout: 5000
 
 production:
-  adapter: sqlite3
+  <<: *default
```
Comment (jcoyne) [inline config/database.yml:24]: Purl prod is load balanced, so you aren't going to be able to use a sqlite database, right?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: config/database.yml (hunk 2 above)
Comment (jcoyne) [reply config/database.yml:24]: I see #744

Comment (thatbudakguy) [reply config/database.yml:24]: Yeah, this defers the solution for prod for the time being.

File: spec/rails_helper.rb (hunk 1 above)
Comment (thatbudakguy) [reply spec/rails_helper.rb:41]: I tried that and it didn't actually clean the database. 🤷🏻 I copied this config from another app where we used this setup, and it did what I wanted.

Comment (jcoyne) [reply spec/rails_helper.rb:41]: There is almost no reason for it in a modern Rails app.  Please don't stuff this in this PR.

**Comment by thatbudakguy** [comment]
superseded by https://github.com/sul-dlss/metrics/pull/6

**Pull request closed without merging**